systest/backup/advanced-scenarios/127-Namespace/backup_test.go

```diff
@@ -0,0 +1,60 @@
+	headerAlpha1 = http.Header{}
+	headerAlpha2 = http.Header{}
+)
+
+const (
+	accessJwtHeader = "X-Dgraph-AccessToken"
+	restoreLocation = "/data/backups/"
+	backupDst       = "/data/backups/"
+)
+
+func Test127PlusNamespaces(t *testing.T) {
+	jwtTokenAlpha1 := testutil.GrootHttpLogin("http://" + testutil.SockAddrHttp + "/admin").AccessJwt
+	headerAlpha1.Set(accessJwtHeader, jwtTokenAlpha1)
+	headerAlpha1.Set("Content-Type", "application/json")
+	jwtTokenAlpha2 := testutil.GrootHttpLogin("http://" + testutil.ContainerAddr("alpha2", 8080) + "/admin").AccessJwt
+	headerAlpha2.Set(accessJwtHeader, jwtTokenAlpha2)
+	headerAlpha2.Set("Content-Type", "application/json")
+	_ = e2eCommon.CreateNamespace(t, headerAlpha1, e2eCommon.CreateNamespaceParams{CustomGraphAdminURLs: "", NamespaceQuant: 20})
+	utilsCommon.AddSchema(t, headerAlpha1, "alpha1")
+	e2eCommon.AssertGetGQLSchema(t, testutil.ContainerAddr("alpha1", 8080), headerAlpha1)
+	utilsCommon.AddData(t, 1, 10, jwtTokenAlpha1, "alpha1")
+	utilsCommon.CheckDataExists(t, 10, jwtTokenAlpha1, "alpha1")
+	_ = e2eCommon.CreateNamespace(t, headerAlpha1, e2eCommon.CreateNamespaceParams{CustomGraphAdminURLs: "", NamespaceQuant: 20})
+	utilsCommon.AddData(t, 11, 20, jwtTokenAlpha1, "alpha1")
+	utilsCommon.CheckDataExists(t, 20, jwtTokenAlpha1, "alpha1")
+	_ = e2eCommon.CreateNamespace(t, headerAlpha1, e2eCommon.CreateNamespaceParams{CustomGraphAdminURLs: "", NamespaceQuant: 20})
+	utilsCommon.AddData(t, 21, 30, jwtTokenAlpha1, "alpha1")
+	utilsCommon.CheckDataExists(t, 30, jwtTokenAlpha1, "alpha1")
+	_ = e2eCommon.CreateNamespace(t, headerAlpha1, e2eCommon.CreateNamespaceParams{CustomGraphAdminURLs: "", NamespaceQuant: 20})
+	utilsCommon.AddData(t, 31, 40, jwtTokenAlpha1, "alpha1")
+	utilsCommon.CheckDataExists(t, 40, jwtTokenAlpha1, "alpha1")
+	_ = e2eCommon.CreateNamespace(t, headerAlpha1, e2eCommon.CreateNamespaceParams{CustomGraphAdminURLs: "", NamespaceQuant: 20})
+	utilsCommon.AddData(t, 41, 50, jwtTokenAlpha1, "alpha1")
+	utilsCommon.CheckDataExists(t, 50, jwtTokenAlpha1, "alpha1")
+	_ = e2eCommon.CreateNamespace(t, headerAlpha1, e2eCommon.CreateNamespaceParams{CustomGraphAdminURLs: "", NamespaceQuant: 20})
+	utilsCommon.AddData(t, 51, 60, jwtTokenAlpha1, "alpha1")
+	utilsCommon.CheckDataExists(t, 60, jwtTokenAlpha1, "alpha1")
+	_ = e2eCommon.CreateNamespace(t, headerAlpha1, e2eCommon.CreateNamespaceParams{CustomGraphAdminURLs: "", NamespaceQuant: 30})
+	utilsCommon.AddData(t, 61, 70, jwtTokenAlpha1, "alpha1")
+	utilsCommon.CheckDataExists(t, 70, jwtTokenAlpha1, "alpha1")
+	utilsCommon.TakeBackup(t, jwtTokenAlpha1, backupDst, "alpha1")
+	utilsCommon.RunRestore(t, jwtTokenAlpha2, restoreLocation, "alpha2")
+	dg := testutil.DgClientWithLogin(t, "groot", "password", x.GalaxyNamespace)
+	testutil.WaitForRestore(t, dg, testutil.ContainerAddr("alpha2", 8080))
+	e2eCommon.AssertGetGQLSchema(t, testutil.ContainerAddr("alpha2", 8080), headerAlpha2)
+	utilsCommon.CheckDataExists(t, 70, jwtTokenAlpha2, "alpha2")
```

Comment:
I don't think we need so many steps, what would more useful would be distributing data across different namespaces as well.

I'd structure it like this:
- Step 1, Create 50 namespaces, add data to namespace 0, take backup and restore and check if data is present. (in the restored backend)
- Step 2. Create another 50 namespaced, add data to namespace 51, take backup and restore and check if the data is present in 0 and 51
- Step 3 Create another 50 namespaces, add data to namespace 130,, take backup and restore and check if data is present in 0, 51 and 130. 


With this we check iteratively adding data to new namespaces preserves data in older namespaces across restores. We also check the namespace increment is monotonic, that is after first restore we get the namespace 51 created.

Reply:
**Resolved**
Now the test has the following flow:-

```
Create 1-50 Namespaces
Add data to Namespace 0
Backup on Alpha1
Restore on Alpha2
Log into Namespace 0 on Alpha2
CheckDataExist
 
Create 51-100 Namespaces
Add data to 51 Namespace
Backup on Alpha1
Restore on Alpha 2
Log into Namespace 0 on Alpha2
CheckDataExist
Log into Namespace51 on Alpha2
CheckDataExist


Create 101-130 Namespaces
Add data to Namespace 130
Backup on Alpha1
Restore on Alpha 2
Log into Namespace 0 on Alpha2
CheckDataExist
Log into Namespace51 on Alpha2
CheckDataExist
Log into Namespace130 on Alpha2
CheckDataExist
```